plugins/home/src/components/CustomHomepage/types.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactElement } from 'react';
+import { Layout } from 'react-grid-layout';
+import { z } from 'zod';
+import { RJSFSchema } from '@rjsf/utils';
+
+const RSJFTypeSchema: z.ZodType<RJSFSchema> = z.any();
+const ReactElementSchema: z.ZodType<ReactElement> = z.any();
+const LayoutSchema: z.ZodType<Layout> = z.any();
```

Comment:
still not sure about this 3 lines - not blocking for me though. Still learning about the purpose of `zod`, but it seems to me like the `component`, `layout` & `settingSchema` shouldn't be part of the zod object at all & rather be added later in the type at the bottom to keep the typing in TypeScript correct (and not overrite it with any). So Zod only extends the limited validation, which TypeScript is capable of.

You could [extend the type](https://www.typescriptlang.org/docs/handbook/2/objects.html#extending-types) or use [intersection](https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types) smth. like the following:

```diff
- export type Widget = z.infer<typeof WidgetSchema>;
+ export type Widget = z.infer<(typeof WidgetSchema) & { settingsSchema?: RJSFSchema }>
```

This way you wouldn't override the type of `settingsSchema` for a widget with the type `Widget` in Typescript with any. And for the `component` the same is valid as mentioned by @freben in `api-report`.

Reply:
I am not too familiar with zod either and found out this way of using external types in it. Also something that could be improved once someone knows how 😁

Reply:
Couldn't really figure out a way to get rid of these for now.